sql/atxs/atxs.go

```diff
@@ -141,6 +142,51 @@ func GetIDByEpochAndNodeID(db sql.Executor, epoch types.EpochID, nodeID types.No
 	return id, err
 }
 
+// GetByEpochAndNodeID gets any ATX by the specified NodeID in the given epoch.
+func GetByEpochAndNodeID(db sql.Executor, epoch types.EpochID, nodeID types.NodeID) (*types.VerifiedActivationTx, error) {
```

Comment:
but what you suggested takes 2 db read to accomplished. this new method just needs 1 db read. that's a worthwhile efficiency IMO.

Reply:
It could still be done in a single transaction with

```golang
db.WithTx(ctx, func(tx *sql.Tx) error {
		atxid, _ := atxs.GetIDByEpochAndNodeID(tx, epoch, nodeId)
		atx, _ := atxs.Get(tx, atxid)
}
```

or would this be less efficient?

Reply:
it's disk reads at the db level. disregarding disk reads for database index, isn't it 2 disk read vs 1 disk read in terms of the actual data?